crates/dbsp/src/storage/file/cache.rs

```diff
@@ -137,15 +137,18 @@ impl BufferCache<FileCacheEntry> {
         compression: Option<Compression>,
         stats: &AtomicCacheStats,
     ) -> Result<FileCacheEntry, Error> {
-        match self.get(file, location, stats) {
-            Some(entry) => Ok(entry),
+        let start = Instant::now();
+        let (access, entry) = match self.get(file, location) {
+            Some(entry) => (CacheAccess::Hit, entry),
             None => {
                 let block = file.read_block(location)?;
                 let entry = FileCacheEntry::from_read(block, location, compression)?;
                 self.insert(file.file_id(), location.offset, entry.clone());
-                Ok(entry)
+                (CacheAccess::Miss, entry)
             }
-        }
+        };
+        stats.record(access, start.elapsed(), location);
```

Comment:
This is something for @abhizer and not for this PR but I don't like we always have to invent something and write code to record all this metrics stuff.. I'd rather we could just use an off-the-shelve crate for it